### PR TITLE
Rename foreign_objects to extra_objects

### PR DESCRIPTION
**Pull request by gridbugs**
Signed-off-by: Stephen Sherratt <stephen@sherra.tt>

### CODE REVIEW

File: src/dune_rules/dune_file.ml [hunk 1]
```diff
@@ -128,7 +128,7 @@ module Buildable = struct
     ; empty_module_interface_if_absent : bool
     ; libraries : Lib_dep.t list
     ; foreign_archives : (Loc.t * Foreign.Archive.t) list
-    ; foreign_objects : Foreign.Objects.t
+    ; extra_objects : Foreign.Objects.t
```
Comment (snowleopard) [inline src/dune_rules/dune_file.ml:131]: This naming mismatch suggests that there is scope for further refactoring. (Not necessarily in this PR.)

**Review by snowleopard — APPROVED** [review]
Looks good to me, thanks!

File: src/dune_rules/dune_file.ml (hunk 1 above)
Comment (gridbugs) [reply src/dune_rules/dune_file.ml:131]: Good point! Think it can just go in a new file (objects.ml)?

File: doc/concepts.rst [hunk 2]
```diff
@@ -1193,7 +1193,7 @@ Foreign Sources, Archives and Objects
 Dune provides basic support for including foreign source files as well
 as archives of foreign object files into OCaml projects via the
 ``foreign_stubs`` and ``foreign_archives`` fields. Individual object
```
Comment (Alizter) [inline doc/concepts.rst:1195]: So stubs and archives stay foreign whereas objects become extra?

Comment (gridbugs) [reply doc/concepts.rst:1195]: That's correct. The `foreign_objects` field was only added recently and we want to rename it before the next release to reflect that the binary object files are not necessarily foreign in the same sense that foreign stubs are foreign (ie. they are files in a non-ocaml source language). The `foreign_archives` field refers to archives of binary files which are also not necessarily foreign, but this field already exists in released versions of dune so we can't rename it easily.

**Review by Alizter — APPROVED** [review]

**Review by christinerose — APPROVED** [review]

**Pull request merged**